src/Microsoft.Extensions.DependencyInjection.Ordered/OrderedEnumerableServiceDescriptor.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection.Ordered
+{
+    public class OrderedEnumerableServiceDescriptor: ServiceDescriptor
```

Comment:
So if container author decides to implement `IOrdered` natively he has a descriptor for it and does not have to search for our private type instance.